diff -u a/iaio.c b/iaio.c
--- a/iaio.c	2009-01-22 16:09:18.000000000 -0800
+++ b/iaio.c	2009-02-07 11:52:26.000000000 -0800
@@ -43,6 +43,7 @@
 
 #include "iaio.h"
 
+#ifdef HAS_SDL
 static inline int iaio_displayimage( iaio_t* iaio, ia_image_t* iar )
 {
     uint32_t i;
@@ -64,6 +65,7 @@
     SDL_UpdateRect( iaio->screen, 0, 0, 0, 0 );
     return 0;
 }
+#endif
 
 static inline int iaio_saveimage ( iaio_t* iaio, ia_image_t* iar )
 {
@@ -87,11 +89,13 @@
         if( iaio_saveimage(iaio, iar) )
             return 1;
     }
+    #ifdef HAS_SDL
     if( iaio->output_type & IAIO_DISPLAY )
     {
         if( iaio_displayimage(iaio, iar) )
             return 1;
     }
+    #endif
     return 0;
 }
 
@@ -624,6 +628,7 @@
     ia_free( iaio->fin.buf );
 }
 
+#ifdef HAS_SDL
 static inline int iaio_display_init( iaio_t* iaio )
 {
     iaio->screen = NULL;
@@ -646,6 +651,7 @@
 {
     SDL_Quit();
 }
+#endif
 
 iaio_t* iaio_open( ia_param_t* p )
 {
@@ -684,11 +690,13 @@
                                      FI_RGBA_RED,FI_RGBA_GREEN,FI_RGBA_BLUE );
 
     }
+    #ifdef HAS_SDL
     if( p->display )
     {
         iaio->output_type |= IAIO_DISPLAY;
         iaio_display_init( iaio );
     }
+    #endif
 
     iaio->eoi = false;
 
@@ -699,8 +707,10 @@
 {
     if( iaio->output_type & IAIO_DISK )
         FreeImage_Unload( iaio->dib );
+    #ifdef HAS_SDL
     if( iaio->output_type & IAIO_DISPLAY )
         iaio_display_close();
+    #endif
 
     if( iaio->input_type & IAIO_CAMERA )
         iaio_cam_close( iaio );
diff -u a/iaio.h b/iaio.h
--- a/iaio.h	2009-01-22 16:09:18.000000000 -0800
+++ b/iaio.h	2009-02-07 11:52:26.000000000 -0800
@@ -23,11 +23,14 @@
 #ifndef _H_IAIO
 #define _H_IAIO
 
-#include <FreeImage.h>
-#include <SDL/SDL.h>
 #include "common.h"
 #include "image_analyzer.h"
 
+#include <FreeImage.h>
+#ifdef HAS_SDL
+#include <SDL/SDL.h>
+#endif
+
 #define IAIO_DISK       1
 #define IAIO_DISPLAY    2
 #define IAIO_CAMERA     3
@@ -71,7 +74,9 @@
     iaio_cam_t      cam;        // for cam input
     iaio_file_t     fin;        // for file input
 
+    #ifdef HAS_SDL
     SDL_Surface*    screen;     // for displaying video
+    #endif
     FIBITMAP*       dib;            // for writing images to disk
     uint64_t    last_frame;
     uint32_t    i_size;
diff -u a/Makefile b/Makefile
--- a/Makefile	2009-01-22 16:09:18.000000000 -0800
+++ b/Makefile	2009-02-07 11:53:21.000000000 -0800
@@ -1,7 +1,7 @@
-#CC		:= /home/jdegges/gumstix/gumstix-oe/tmp/cross/arm-angstrom-linux-gnueabi/bin/gcc
-CC		:= gcc
-CFLAGS	:= --fast-math -O6 -g -W -Wall
-LIBS	:= -lm -lfreeimage -lpthread -lSDL
+CC		:= ${GUMSTIXTOP}/tmp/cross/arm-angstrom-linux-gnueabi/bin/gcc
+#CC		:= gcc
+CFLAGS	:= --fast-math -O6
+LIBS	:= -lm -lfreeimage -lpthread
 OBJECTS	:= $(patsubst %.c,%.o,$(wildcard *.c filters/*.c))
 
 ia : $(OBJECTS)
